<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Session 03 - Exercises Key</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2022 Module 15</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="Session01_practical.html">Session 1</a>
    </li>
    <li>
      <a href="Session02_practical.html">Session 2</a>
    </li>
    <li>
      <a href="Session03_practical.html">Session 3</a>
    </li>
    <li>
      <a href="SISGM15_prac4.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="SISGM15_prac5.html">Session 5</a>
    </li>
    <li>
      <a href="SISGM15_prac6.html">Session 6</a>
    </li>
    <li>
      <a href="Session07_practical.html">Session 7</a>
    </li>
    <li>
      <a href="Session08_practical.html">Session 8</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="SISGM15_prac9.html">Session 9</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Session 03 - Exercises Key</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SISG2022_Association_Mapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/joelle.mbatchou/software/github/SISG2022_Association_Mapping/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2022AssociationMappingtreecb3e60480f3b684a3cc48e202d90ce2f14e9d7e2targetblankcb3e604a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/cb3e60480f3b684a3cc48e202d90ce2f14e9d7e2" target="_blank">cb3e604</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2022AssociationMappingtreecb3e60480f3b684a3cc48e202d90ce2f14e9d7e2targetblankcb3e604a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/cb3e60480f3b684a3cc48e202d90ce2f14e9d7e2" target="_blank">cb3e604</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    lectures/.DS_Store

Untracked files:
    Untracked:  analysis/Session07_practical_Key.Rmd
    Untracked:  analysis/Session08_practical_Key.Rmd
    Untracked:  data/sim_rels_geno.bed
    Untracked:  data/sim_rels_geno.bed?dl=0

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Session03_practical_Key.Rmd</code>) and HTML
(<code>docs/Session03_practical_Key.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/cb3e60480f3b684a3cc48e202d90ce2f14e9d7e2/analysis/Session03_practical_Key.Rmd" target="_blank">cb3e604</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
add key
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<ul>
<li>Make sure that R is installed on your computer</li>
<li>For this lab, we will use the following R libraries:</li>
</ul>
<pre class="r"><code>require(data.table)
require(dplyr)
require(tidyr)
require(GWASTools)
require(ggplot2)</code></pre>
<div id="gwas-in-samples-with-structure" class="section level2">
<h2>GWAS in Samples with Structure</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>We will be analyzing a simulated data set which contains sample
structure to better understand the impact it can have in GWAS analyses
if not accounted for. We will perform GWAS on a quantitative phenotype
which was simulated to have high heritability and be highly
polygenic.</p>
<p>The file <a
href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/master/data">“sim_rels_pheno.txt”</a>”
contains the phenotype measurements for a set of individuals and the
file “sim_rels_geno.bed” is a binary file in PLINK BED format with
accompanying BIM and FAM files which contains the genotype data
<strong>at null variants</strong> (i.e. simulated as not associated with
the phenotype).<br />
<em>How should we expect the QQ/Manhatthan plots to look like under this
scenario?</em></p>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>Here are some things to try:</p>
<ol style="list-style-type: decimal">
<li>Examine the dataset:</li>
</ol>
<ul>
<li>How many samples are present?</li>
</ul>
<pre class="r"><code>famfile &lt;- fread(&quot;data/sim_rels_geno.fam&quot;, header = FALSE)
famfile %&gt;% str</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  2400 obs. of  6 variables:
 $ V1: int  2307 379 478 1545 990 1907 369 1694 2137 2314 ...
 $ V2: int  2307 379 478 1545 990 1907 369 1694 2137 2314 ...
 $ V3: int  0 0 0 0 0 0 0 0 0 0 ...
 $ V4: int  0 0 0 0 0 0 0 0 0 0 ...
 $ V5: int  1 2 1 1 1 2 2 1 2 1 ...
 $ V6: int  -9 -9 -9 -9 -9 -9 -9 -9 -9 -9 ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<ul>
<li>How many SNPs? In how many chromosomes?</li>
</ul>
<pre class="r"><code>bimfile &lt;- fread(&quot;data/sim_rels_geno.bim&quot;, header = FALSE)
bimfile %&gt;% str</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  106134 obs. of  6 variables:
 $ V1: int  1 1 1 1 1 1 1 1 1 1 ...
 $ V2: chr  &quot;1:12000011:A:C&quot; &quot;1:12000012:A:C&quot; &quot;1:12000019:T:C&quot; &quot;1:12000027:C:T&quot; ...
 $ V3: int  0 0 0 0 0 0 0 0 0 0 ...
 $ V4: int  12000011 12000012 12000019 12000027 12000036 12000061 12000073 12000074 12000117 12000136 ...
 $ V5: chr  &quot;A&quot; &quot;A&quot; &quot;T&quot; &quot;C&quot; ...
 $ V6: chr  &quot;C&quot; &quot;C&quot; &quot;C&quot; &quot;T&quot; ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<pre class="r"><code>bimfile %&gt;% select(V1) %&gt;% table</code></pre>
<pre><code>.
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
4918 4857 4813 4772 4810 4914 4840 4696 4790 4906 4782 4756 4803 4671 4814 4869 
  17   18   19   20   21   22 
4632 4834 4908 4942 4947 4860 </code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Examine the phenotype data:</li>
</ol>
<ul>
<li>How many individuals in the study have measurements?</li>
</ul>
<pre class="r"><code>yfile &lt;- fread(&quot;data/sim_rels_pheno.txt&quot;, header = TRUE)
yfile %&gt;% str</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  2400 obs. of  3 variables:
 $ FID  : int  2307 379 478 1545 990 1907 369 1694 2137 2314 ...
 $ IID  : int  2307 379 478 1545 990 1907 369 1694 2137 2314 ...
 $ Pheno: num  0.00999 -1.45253 0.11097 1.11363 -0.20993 ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<pre class="r"><code>yfile %&gt;% pull(Pheno) %&gt;% is.na %&gt;% table</code></pre>
<pre><code>.
FALSE 
 2400 </code></pre>
<ul>
<li>Make a visual of the distribution of the phenotype?</li>
</ul>
<pre class="r"><code>yfile %&gt;%
  ggplot(aes(x = Pheno)) +
  geom_histogram(colour=&quot;black&quot;, fill=&quot;white&quot;)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/Session03_practical_Key.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="3" style="list-style-type: decimal">
<li>Using PLINK, perform a GWAS using the phenotype file
<code>sim_rels_pheno.txt</code> and the
<code>sim_rels_geno.{bed,bim,fam}</code> genotype files. Only perform
association test on SNPs that pass the following quality control
threshold filters:</li>
</ol>
<ul>
<li>minor allele frequency (MAF) &gt; 0.01</li>
<li>at least a 99% genotyping call rate (less than 1% missing)</li>
<li>HWE p-values greater than 0.001</li>
</ul>
<pre class="bash"><code>~/software/bins/plink2 --bfile data/sim_rels_geno --pheno data/sim_rels_pheno.txt --pheno-name Pheno --maf 0.01 --geno 0.01 --hwe 0.001 --autosome --glm allow-no-covars --out /tmp/gwas_plink</code></pre>
<pre><code>PLINK v2.00a3 AVX2 (12 Dec 2020)               www.cog-genomics.org/plink/2.0/
(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /tmp/gwas_plink.log.
Options in effect:
  --autosome
  --bfile data/sim_rels_geno
  --geno 0.01
  --glm allow-no-covars
  --hwe 0.001
  --maf 0.01
  --out /tmp/gwas_plink
  --pheno data/sim_rels_pheno.txt
  --pheno-name Pheno

Start time: Mon Jul 25 16:18:13 2022
16384 MiB RAM detected; reserving 8192 MiB for main workspace.
Using up to 12 threads (change this with --threads).
2400 samples (1179 females, 1221 males; 2400 founders) loaded from
data/sim_rels_geno.fam.
106134 variants loaded from data/sim_rels_geno.bim.
1 quantitative phenotype loaded (2400 values).
Calculating allele frequencies... 0%61%done.
--geno: 0 variants removed due to missing genotype data.
--hwe: 123 variants removed due to Hardy-Weinberg exact test (founders only).
125 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
105886 variants remaining after main filters.
--glm linear regression on phenotype &#39;Pheno&#39;: 0%61%done.
Results written to /tmp/gwas_plink.Pheno.glm.linear .
End time: Mon Jul 25 16:18:13 2022</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Make a Manhattan plot of the association results using the
<code>manhattanPlot()</code> R function.</li>
</ol>
<pre class="r"><code>plink.gwas &lt;- fread(&quot;/tmp/gwas_plink.Pheno.glm.linear&quot;, header = TRUE)
plink.gwas %&gt;% str</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  105886 obs. of  13 variables:
 $ #CHROM : int  1 1 1 1 1 1 1 1 1 1 ...
 $ POS    : int  12000011 12000012 12000019 12000027 12000036 12000061 12000073 12000074 12000117 12000136 ...
 $ ID     : chr  &quot;1:12000011:A:C&quot; &quot;1:12000012:A:C&quot; &quot;1:12000019:T:C&quot; &quot;1:12000027:C:T&quot; ...
 $ REF    : chr  &quot;C&quot; &quot;C&quot; &quot;C&quot; &quot;T&quot; ...
 $ ALT    : chr  &quot;A&quot; &quot;A&quot; &quot;T&quot; &quot;C&quot; ...
 $ A1     : chr  &quot;A&quot; &quot;A&quot; &quot;T&quot; &quot;C&quot; ...
 $ TEST   : chr  &quot;ADD&quot; &quot;ADD&quot; &quot;ADD&quot; &quot;ADD&quot; ...
 $ OBS_CT : int  2400 2400 2400 2400 2400 2400 2400 2400 2400 2400 ...
 $ BETA   : num  0.0122 -0.018 -0.0849 0.0125 0.0111 ...
 $ SE     : num  0.0438 0.0362 0.0284 0.0435 0.0288 ...
 $ T_STAT : num  0.279 -0.497 -2.992 0.288 0.387 ...
 $ P      : num  0.78 0.6192 0.0028 0.7731 0.6991 ...
 $ ERRCODE: chr  &quot;.&quot; &quot;.&quot; &quot;.&quot; &quot;.&quot; ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<pre class="r"><code>manhattanPlot(
  p = plink.gwas$P,
  chromosome = plink.gwas$`#CHROM`, 
  thinThreshold = 1e-4,
  main= &quot;Manhattan plot of GWAS with PLINK&quot;
)</code></pre>
<p><img src="figure/Session03_practical_Key.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="5" style="list-style-type: decimal">
<li>Make a Q-Q plot of the association results using the
<code>qqPlot()</code> R function.</li>
</ol>
<pre class="r"><code>qqPlot(
  pval = plink.gwas$P,
  thinThreshold = 1e-4,
  main= &quot;Q-Q plot of GWAS with PLINK&quot;
 )</code></pre>
<p><img src="figure/Session03_practical_Key.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="6" style="list-style-type: decimal">
<li>Compute the genomic control inflation factor <span
class="math inline">\(\lambda_{GC}\)</span> based on the p-values. Is
there evidence of possible inflation due to confounding?</li>
</ol>
<pre class="r"><code>chisq.stats &lt;- qchisq(plink.gwas$P, df = 1, lower.tail = FALSE)
median(chisq.stats) / qchisq(0.5,1)</code></pre>
<pre><code>[1] 1.148451</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Now use REGENIE to perform a GWAS of the phenotype using a whole
genome regression model.</li>
</ol>
<ul>
<li>We want to use high quality variants in the Step 1 null model
fitting. Using PLINK, apply QC filters to remove variants with MAF below
5%, missingness above 1%, HWE p-value below 0.001, minor allele count
(MAC) below 20.</li>
</ul>
<pre class="bash"><code>~/software/bins/plink2 --bfile data/sim_rels_geno --maf 0.05 --geno 0.01 --hwe 0.001 --mac 20 --write-snplist --out /tmp/qc_pass</code></pre>
<pre><code>PLINK v2.00a3 AVX2 (12 Dec 2020)               www.cog-genomics.org/plink/2.0/
(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /tmp/qc_pass.log.
Options in effect:
  --bfile data/sim_rels_geno
  --geno 0.01
  --hwe 0.001
  --mac 20
  --maf 0.05
  --out /tmp/qc_pass
  --write-snplist

Start time: Mon Jul 25 16:18:28 2022
16384 MiB RAM detected; reserving 8192 MiB for main workspace.
Using up to 12 threads (change this with --threads).
2400 samples (1179 females, 1221 males; 2400 founders) loaded from
data/sim_rels_geno.fam.
106134 variants loaded from data/sim_rels_geno.bim.
Note: No phenotype data present.
Calculating allele frequencies... 0%61%done.
--geno: 0 variants removed due to missing genotype data.
--hwe: 123 variants removed due to Hardy-Weinberg exact test (founders only).
8624 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
97387 variants remaining after main filters.
--write-snplist: Variant IDs written to /tmp/qc_pass.snplist .
End time: Mon Jul 25 16:18:28 2022</code></pre>
<ul>
<li>Run REGENIE Step 1 to fit the null model and obtain polygenic
predictions using a leave-one-chromosome-out (LOCO) scheme</li>
</ul>
<pre class="bash"><code>regenie --bed data/sim_rels_geno --phenoFile data/sim_rels_pheno.txt --step 1 --loocv --bsize 1000 --qt --extract /tmp/qc_pass.snplist --out /tmp/regenie_step1</code></pre>
<p>The prediction list file output from Step 1 contains the path to the
LOCO polygenic predictions:</p>
<pre class="bash"><code>cat /tmp/regenie_step1_pred.list</code></pre>
<pre><code>Pheno /tmp/regenie_step1_1.loco</code></pre>
<ul>
<li>Run REGENIE Step 2 to perform association testing <strong>at the
same set of SNPs tested in PLINK</strong>.</li>
</ul>
<pre class="r"><code>plink.gwas %&gt;%
  select(ID) %&gt;%
  fwrite(&quot;/tmp/plink_gwas.snplist&quot;, col.names = FALSE, quote = FALSE)</code></pre>
<pre class="bash"><code>regenie --bed data/sim_rels_geno --phenoFile data/sim_rels_pheno.txt --step 2 --bsize 400 --qt --pred /tmp/regenie_step1_pred.list --extract /tmp/plink_gwas.snplist --out /tmp/regenie_step2</code></pre>
<ul>
<li>Generate Manhatthan and Q-Q plots based on the association results
and compute <span class="math inline">\(\lambda_{GC}\)</span>. Compare
with output from Questions 4-6.</li>
</ul>
<pre class="r"><code>regenie.gwas &lt;- fread(&quot;/tmp/regenie_step2_Pheno.regenie&quot;, header = TRUE)
regenie.gwas %&gt;% str</code></pre>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  105886 obs. of  13 variables:
 $ CHROM  : int  1 1 1 1 1 1 1 1 1 1 ...
 $ GENPOS : int  12000011 12000012 12000019 12000027 12000036 12000061 12000073 12000074 12000117 12000136 ...
 $ ID     : chr  &quot;1:12000011:A:C&quot; &quot;1:12000012:A:C&quot; &quot;1:12000019:T:C&quot; &quot;1:12000027:C:T&quot; ...
 $ ALLELE0: chr  &quot;C&quot; &quot;C&quot; &quot;C&quot; &quot;T&quot; ...
 $ ALLELE1: chr  &quot;A&quot; &quot;A&quot; &quot;T&quot; &quot;C&quot; ...
 $ A1FREQ : num  0.12 0.187 0.402 0.12 0.415 ...
 $ N      : int  2400 2400 2400 2400 2400 2400 2400 2400 2400 2400 ...
 $ TEST   : chr  &quot;ADD&quot; &quot;ADD&quot; &quot;ADD&quot; &quot;ADD&quot; ...
 $ BETA   : num  0.00851 -0.01943 -0.0747 -0.023 0.01463 ...
 $ SE     : num  0.0419 0.0346 0.0272 0.0416 0.0275 ...
 $ CHISQ  : num  0.0413 0.3153 7.5548 0.3058 0.2823 ...
 $ LOG10P : num  0.0762 0.2407 2.2229 0.2364 0.2254 ...
 $ EXTRA  : logi  NA NA NA NA NA NA ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<pre class="r"><code>manhattanPlot(
  p = 10^-regenie.gwas$LOG10P,
  chromosome = regenie.gwas$CHROM, 
  thinThreshold = 1e-4,
  main= &quot;Manhattan plot of GWAS with REGENIE&quot;
)</code></pre>
<p><img src="figure/Session03_practical_Key.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>qqPlot(
  pval = 10^-regenie.gwas$LOG10P,
  thinThreshold = 1e-4,
  main= &quot;Q-Q plot of GWAS with REGENIE&quot;
 )</code></pre>
<p><img src="figure/Session03_practical_Key.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chisq.stats &lt;- qchisq(10^-regenie.gwas$LOG10P, df = 1, lower.tail = FALSE)
median(chisq.stats) / qchisq(0.5,1)</code></pre>
<pre><code>[1] 0.9962878</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] ggplot2_3.3.3       GWASTools_1.32.0    Biobase_2.46.0     
[4] BiocGenerics_0.32.0 tidyr_1.0.2         dplyr_1.0.8        
[7] data.table_1.13.2   workflowr_1.7.0    

loaded via a namespace (and not attached):
 [1] nlme_3.1-140         fs_1.5.2             bit64_4.0.5         
 [4] httr_1.4.3           rprojroot_2.0.3      tools_3.6.1         
 [7] backports_1.1.5      bslib_0.3.1          utf8_1.1.4          
[10] R6_2.4.1             colorspace_2.0-0     DBI_1.1.3           
[13] mgcv_1.8-28          withr_2.5.0          DNAcopy_1.60.0      
[16] tidyselect_1.1.1     processx_3.5.3       bit_4.0.4           
[19] compiler_3.6.1       git2r_0.30.1         cli_3.1.1           
[22] quantreg_5.54        mice_3.8.0           SparseM_1.78        
[25] sandwich_3.0-1       labeling_0.4.2       sass_0.4.0          
[28] scales_1.1.1         lmtest_0.9-40        quantsmooth_1.52.0  
[31] callr_3.7.0          stringr_1.4.0        digest_0.6.25       
[34] minqa_1.2.4          GWASExactHW_1.01     rmarkdown_2.14      
[37] pkgconfig_2.0.3      htmltools_0.5.2      lme4_1.1-21         
[40] fastmap_1.1.0        highr_0.8            rlang_1.0.4         
[43] rstudioapi_0.13      RSQLite_2.2.15       farver_2.0.3        
[46] jquerylib_0.1.4      generics_0.0.2       zoo_1.8-9           
[49] jsonlite_1.7.2       magrittr_1.5         Matrix_1.2-17       
[52] Rcpp_1.0.8.3         munsell_0.5.0        fansi_0.4.1         
[55] lifecycle_1.0.1      stringi_1.4.6        whisker_0.4         
[58] yaml_2.2.1           MASS_7.3-51.4        grid_3.6.1          
[61] formula.tools_1.7.1  blob_1.2.3           promises_1.2.0.1    
[64] crayon_1.3.4         lattice_0.20-38      splines_3.6.1       
[67] knitr_1.39           ps_1.7.0             pillar_1.7.0        
[70] boot_1.3-22          logistf_1.24.1       gdsfmt_1.22.0       
[73] glue_1.6.1           evaluate_0.15        getPass_0.2-2       
[76] operator.tools_1.6.3 vctrs_0.3.8          nloptr_1.2.2.1      
[79] httpuv_1.6.5         MatrixModels_0.4-1   gtable_0.3.0        
[82] purrr_0.3.3          cachem_1.0.6         xfun_0.31           
[85] broom_0.5.5          later_1.3.0          survival_3.2-10     
[88] tibble_3.1.6         memoise_2.0.1        ellipsis_0.3.2      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
