<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Session 07 - Exercises Key</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2022 Module 15</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="Session01_practical.html">Session 1</a>
    </li>
    <li>
      <a href="Session02_practical.html">Session 2</a>
    </li>
    <li>
      <a href="Session03_practical.html">Session 3</a>
    </li>
    <li>
      <a href="SISGM15_prac4.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="SISGM15_prac5.html">Session 5</a>
    </li>
    <li>
      <a href="SISGM15_prac6.html">Session 6</a>
    </li>
    <li>
      <a href="Session07_practical.html">Session 7</a>
    </li>
    <li>
      <a href="Session08_practical.html">Session 8</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="SISGM15_prac9.html">Session 9</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Session 07 - Exercises Key</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SISG2022_Association_Mapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2022AssociationMappingtreec4a403276457054895323d1abb355f596feee32etargetblankc4a4032a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/c4a403276457054895323d1abb355f596feee32e" target="_blank">c4a4032</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2022AssociationMappingtreec4a403276457054895323d1abb355f596feee32etargetblankc4a4032a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/c4a403276457054895323d1abb355f596feee32e" target="_blank">c4a4032</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    lectures/.DS_Store

Untracked files:
    Untracked:  analysis/.Session07_practical_Key.Rmd.swp
    Untracked:  analysis/SISGM15_prac9Solution.Rmd
    Untracked:  analysis/Session08_practical_Key.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Session07_practical_Key.Rmd</code>) and HTML (<code>docs/Session07_practical_Key.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/c4a403276457054895323d1abb355f596feee32e/analysis/Session07_practical_Key.Rmd" target="_blank">c4a4032</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-26
</td>
<td>
add key
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2022_Association_Mapping/ee9d8e9ff977968e511fc054d2ecfecaa68b2983/docs/Session07_practical_Key.html" target="_blank">ee9d8e9</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/5f9160ce1ecf83193f2a9fb41fa09c16f75d332f/analysis/Session07_practical_Key.Rmd" target="_blank">5f9160c</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-26
</td>
<td>
add key
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/7d4cde5e5ebe4f65a372e61f50bef0ba6e6ad2df/analysis/Session07_practical_Key.Rmd" target="_blank">7d4cde5</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
edit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2022_Association_Mapping/2715c979af966e1976c1836ccac1f33e70816a44/docs/Session07_practical_Key.html" target="_blank">2715c97</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/f3eaf629c763cd32df4580910931d5178f27711c/analysis/Session07_practical_Key.Rmd" target="_blank">f3eaf62</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
fix files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<ul>
<li>Make sure that R is installed on your computer</li>
<li>For this lab, we will use the following R libraries:</li>
</ul>
<pre class="r"><code>require(data.table)
require(dplyr)
require(tidyr)
require(BEDMatrix)
require(SKAT)
require(ACAT)
require(ggplot2)</code></pre>
<div id="rare-variant-analysis" class="section level2">
<h2>Rare Variant Analysis</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>We will look into a dataset collected on a quantitative phenotype which was first analyzed through GWAS and a signal was detected in chromosome 1. Let’s determine whether the signal is present when we focus on rare variation at the locus. In our analyses, <em>we will define rare variants as those with</em> <span class="math inline">\(MAF \leq 5\%\)</span>.</p>
<p>The file <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/master/data">“rv_pheno.txt”</a>” contains the phenotype measurements for a set of individuals and the file “rv_geno_chr1.bed” is a binary file in PLINK BED format with accompanying BIM and FAM files which contains the genotype data.</p>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>Here are some things to try:</p>
<ol style="list-style-type: decimal">
<li>Using PLINK, extract <strong>rare variants</strong> in a new PLINK BED file.</li>
</ol>
<pre class="r"><code>system(&quot;/data/SISG2022M15/exe/plink2 --bfile /data/SISG2022M15/data/rv_geno_chr1 --max-maf 0.05 --maj-ref force --make-bed --out chr1_region_rv&quot;)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Load the data in R:</li>
</ol>
<ul>
<li>Read in the SNPs using R function <code>BEDMatrix()</code></li>
</ul>
<pre class="r"><code>G &lt;- BEDMatrix(&quot;chr1_region_rv&quot;, simple_names = TRUE)</code></pre>
<pre><code>Extracting number of samples and rownames from chr1_region_rv.fam...</code></pre>
<pre><code>Extracting number of variants and colnames from chr1_region_rv.bim...</code></pre>
<ul>
<li>Load the phenotype data from <code>rv_pheno.txt</code></li>
</ul>
<pre class="r"><code>y &lt;- fread(&quot;/data/SISG2022M15/data/rv_pheno.txt&quot;, header = TRUE)</code></pre>
<ul>
<li>Keep only samples who are present both in the genotype as well as phenotype data and who don’t have missing values for the phenotype</li>
</ul>
<pre class="r"><code>ids.keep &lt;- y %&gt;% drop_na(Pheno) %&gt;% pull(IID)
length(ids.keep)</code></pre>
<pre><code>[1] 9949</code></pre>
<pre class="r"><code>G &lt;- G[match(ids.keep, rownames(G)), ]
y &lt;- y %&gt;% drop_na(Pheno)
dim(G)</code></pre>
<pre><code>[1] 9949   56</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Examine the genotype data:</li>
</ol>
<ul>
<li>Compute the minor allele frequency (MAF) for each SNP and plot histogram.</li>
</ul>
<pre class="r"><code>maf &lt;- apply(G, 2, function(x) mean(x, na.rm=TRUE))/2
maf %&gt;% hist(xlab = &quot;Minor allele frequencies&quot;, main = &quot;Distribution of MAF&quot;)</code></pre>
<p><img src="figure/Session07_practical_Key.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/2715c979af966e1976c1836ccac1f33e70816a44/docs/figure/Session07_practical_Key.Rmd/unnamed-chunk-6-1.png" target="_blank">2715c97</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Check for missing values.</li>
</ul>
<pre class="r"><code>sum(is.na(G))</code></pre>
<pre><code>[1] 0</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Run the single variant association tests in PLINK (only for the extracted variants).</li>
</ol>
<pre class="r"><code>system(&quot;/data/SISG2022M15/exe/plink2 --bfile chr1_region_rv --pheno /data/SISG2022M15/data/rv_pheno.txt --pheno-name Pheno --glm allow-no-covars --out sv_test&quot;)</code></pre>
<ul>
<li>What would be your significance threshold after applying Bonferroni correction for the multiple tests (assume the significance level is 0.05)? Is anything significant after this correction?</li>
</ul>
<pre class="r"><code>sv_pvals &lt;- fread(&quot;sv_test.Pheno.glm.linear&quot;)
bonf.p &lt;- 0.05 / length(sv_pvals$P)
bonf.p</code></pre>
<pre><code>[1] 0.0008928571</code></pre>
<pre class="r"><code>sv_pvals[P &lt;= bonf.p, ] %&gt;% 
  arrange(P)</code></pre>
<pre><code>   #CHROM      POS             ID REF ALT A1 TEST OBS_CT      BETA        SE
1:      1 12639385 1:12639385:G:A   A   G  G  ADD   9949  0.391421 0.0955243
2:      1 12057950 1:12057950:C:T   T   C  C  ADD   9949 -0.323574 0.0801934
3:      1 12734720 1:12734720:A:C   C   A  A  ADD   9949 -0.278236 0.0769709
4:      1 12405413 1:12405413:T:C   C   T  T  ADD   9949  0.359249 0.1019850
5:      1 12360016 1:12360016:G:A   A   G  G  ADD   9949 -0.310260 0.0898433
6:      1 12183493 1:12183493:G:A   A   G  G  ADD   9949  0.274311 0.0814842
     T_STAT           P ERRCODE
1:  4.09761 4.20759e-05       .
2: -4.03492 5.50288e-05       .
3: -3.61482 3.02042e-04       .
4:  3.52258 4.29282e-04       .
5: -3.45335 5.55975e-04       .
6:  3.36643 7.64368e-04       .</code></pre>
<ul>
<li>Make a volcano plot (i.e. log10 p-values vs effect sizes).</li>
</ul>
<pre class="r"><code>sv_pvals %&gt;%
  ggplot(aes(x = sv_pvals$BETA, y = -log10(sv_pvals$P))) +
  geom_point() +
  labs(x = &quot;Effect size&quot;, y = &quot;-log10P&quot;)</code></pre>
<p><img src="figure/Session07_practical_Key.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/2715c979af966e1976c1836ccac1f33e70816a44/docs/figure/Session07_practical_Key.Rmd/unnamed-chunk-10-1.png" target="_blank">2715c97</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ol start="5" style="list-style-type: decimal">
<li>We will first compare three collapsing/burden approaches:</li>
</ol>
<ul>
<li>CAST (Binary collapsing approach): for each individual, count where they have a rare allele at any of the sites</li>
<li>MZ Test/GRANVIL (Count based collapsing): for each individual, count the total number of sites where a rare allele is present</li>
<li>Weighted burden test: for each individual, take a weighted count of the rare alleles across sites (for the weights, use <code>weights &lt;- dbeta(MAF, 1, 25)</code>)</li>
</ul>
<p>For each approach, first generate the burden scores vector then test it for association with the phenotype using <code>lm()</code> R function.</p>
<pre class="r"><code># CAST
# count number of rare alleles for each person and determine if it is &gt; 0
burden.cast &lt;- as.numeric( apply(G, 1, sum) &gt; 0 )
lm(y$Pheno ~ burden.cast) %&gt;% summary</code></pre>
<pre><code>
Call:
lm(formula = y$Pheno ~ burden.cast)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.9531 -0.6880  0.0012  0.6822  3.6581 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept) 0.002423   0.015317   0.158    0.874
burden.cast 0.017757   0.020422   0.870    0.385

Residual standard error: 1.01 on 9947 degrees of freedom
Multiple R-squared:  7.6e-05,   Adjusted R-squared:  -2.452e-05 
F-statistic: 0.7561 on 1 and 9947 DF,  p-value: 0.3846</code></pre>
<pre class="r"><code># MZ
# count number of sites with rare alleles for each person
burden.mz &lt;- apply( G &gt; 0 , 1, sum)
lm(y$Pheno ~ burden.mz) %&gt;% summary</code></pre>
<pre><code>
Call:
lm(formula = y$Pheno ~ burden.mz)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.9521 -0.6894 -0.0013  0.6805  3.6591 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept) 0.001444   0.013700   0.105    0.916
burden.mz   0.013492   0.011346   1.189    0.234

Residual standard error: 1.01 on 9947 degrees of freedom
Multiple R-squared:  0.0001421, Adjusted R-squared:  4.162e-05 
F-statistic: 1.414 on 1 and 9947 DF,  p-value: 0.2344</code></pre>
<pre class="r"><code># Weighted burden
# weighted sum of genotype counts across sites
weights &lt;- dbeta(maf, 1, 25) 
burden.weighted &lt;- G %*% weights
lm(y$Pheno ~ burden.weighted) %&gt;% summary</code></pre>
<pre><code>
Call:
lm(formula = y$Pheno ~ burden.weighted)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.9519 -0.6896 -0.0014  0.6804  3.6593 

Coefficients:
                 Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)     0.0012244  0.0136778   0.090    0.929
burden.weighted 0.0006577  0.0005402   1.217    0.223

Residual standard error: 1.01 on 9947 degrees of freedom
Multiple R-squared:  0.000149,  Adjusted R-squared:  4.846e-05 
F-statistic: 1.482 on 1 and 9947 DF,  p-value: 0.2235</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Now use SKAT to test for an association.</li>
</ol>
<pre class="r"><code># fit null model (no covariates)
skat.null &lt;- SKAT_Null_Model(y$Pheno ~ 1 , out_type = &quot;C&quot;)
# Run SKAT association test
SKAT(G, skat.null )$p.value</code></pre>
<pre><code>[1] 8.745405e-11</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Run the omnibus SKAT, but consider setting <span class="math inline">\(\rho\)</span> (i.e.<code>r.corr</code>) to 0 and then 1.</li>
</ol>
<ul>
<li>Compare the results to using the CAST,MZ/GRANVIL and Weighted burden collapsing approaches in Question 5 as well as SKAT in Question 6. What tests do these <span class="math inline">\(\rho\)</span> values correspond to?</li>
</ul>
<pre class="r"><code># Run SKATO association test specifying rho
p.skato.r0 &lt;- SKAT(G, skat.null, r.corr = 0)$p.value
p.skato.r1 &lt;- SKAT(G, skat.null, r.corr = 1)$p.value
c(rho0 = p.skato.r0, rho1 = p.skato.r1)</code></pre>
<pre><code>        rho0         rho1 
8.745405e-11 2.234603e-01 </code></pre>
<ol start="8" style="list-style-type: decimal">
<li>Now the omnibus version of SKAT, but use the “optimal.adj” approach which searches across a range of rho values.</li>
</ol>
<pre class="r"><code># Run SKATO association test using grid of rho values
SKAT(G, skat.null, method=&quot;optimal.adj&quot;)$p.value</code></pre>
<pre><code>[1] 6.121784e-10</code></pre>
<ol start="9" style="list-style-type: decimal">
<li>Run ACATV on the single variant p-values.</li>
</ol>
<pre class="r"><code># `weights` vector is from Qesution 5
acat.weights &lt;- weights^2 * maf * (1 - maf)
p.acatv &lt;- ACAT(sv_pvals$P, weights = acat.weights)
p.acatv</code></pre>
<pre><code>[1] 0.00112117</code></pre>
<ol start="10" style="list-style-type: decimal">
<li>Run ACATO combining the SKAT and BURDEN p-values (from Question 7) with the ACATV p-value.</li>
</ol>
<pre class="r"><code>ACAT( c(p.skato.r0, p.skato.r1, p.acatv) )</code></pre>
<pre><code>[1] 2.623621e-10</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggplot2_3.3.6     ACAT_0.91         SKAT_2.2.4        RSpectra_0.16-1  
 [5] SPAtest_3.1.2     Matrix_1.4-1      BEDMatrix_2.0.3   tidyr_1.2.0      
 [9] dplyr_1.0.9       data.table_1.14.2 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2 xfun_0.31        bslib_0.3.1      purrr_0.3.4     
 [5] lattice_0.20-45  colorspace_2.0-3 vctrs_0.4.1      generics_0.1.3  
 [9] crochet_2.3.0    htmltools_0.5.2  yaml_2.3.5       utf8_1.2.2      
[13] rlang_1.0.3      later_1.3.0      pillar_1.7.0     jquerylib_0.1.4 
[17] withr_2.5.0      glue_1.6.2       DBI_1.1.3        lifecycle_1.0.1 
[21] stringr_1.4.0    munsell_0.5.0    gtable_0.3.0     evaluate_0.15   
[25] labeling_0.4.2   knitr_1.39       callr_3.7.0      fastmap_1.1.0   
[29] httpuv_1.6.5     ps_1.7.1         fansi_1.0.3      highr_0.9       
[33] Rcpp_1.0.9       scales_1.2.0     promises_1.2.0.1 jsonlite_1.8.0  
[37] farver_2.1.1     fs_1.5.2         digest_0.6.29    stringi_1.7.6   
[41] processx_3.7.0   getPass_0.2-2    rprojroot_2.0.3  grid_4.2.1      
[45] cli_3.3.0        tools_4.2.1      magrittr_2.0.3   sass_0.4.1      
[49] tibble_3.1.7     crayon_1.5.1     whisker_0.4      pkgconfig_2.0.3 
[53] ellipsis_0.3.2   assertthat_0.2.1 rmarkdown_2.14   httr_1.4.3      
[57] rstudioapi_0.13  R6_2.5.1         git2r_0.30.1     compiler_4.2.1  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
