<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Session 02 - Exercises Key</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SISG 2022 Module 15</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Sessions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Day 1</li>
    <li>
      <a href="Session01_practical.html">Session 1</a>
    </li>
    <li>
      <a href="Session02_practical.html">Session 2</a>
    </li>
    <li>
      <a href="Session03_practical.html">Session 3</a>
    </li>
    <li>
      <a href="SISGM15_prac4.html">Session 4</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 2</li>
    <li>
      <a href="SISGM15_prac5.html">Session 5</a>
    </li>
    <li>
      <a href="SISGM15_prac6.html">Session 6</a>
    </li>
    <li>
      <a href="Session07_practical.html">Session 7</a>
    </li>
    <li>
      <a href="Session08_practical.html">Session 8</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Day 3</li>
    <li>
      <a href="SISGM15_prac9.html">Session 9</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Session 02 - Exercises Key</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SISG2022_Association_Mapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220530code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220530code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220530)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2022AssociationMappingtree6f1ab71addc277eb8d8119ce3ac73329d5f7815ftargetblank6f1ab71a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/6f1ab71addc277eb8d8119ce3ac73329d5f7815f" target="_blank">6f1ab71</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjoellembatchouSISG2022AssociationMappingtree6f1ab71addc277eb8d8119ce3ac73329d5f7815ftargetblank6f1ab71a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/6f1ab71addc277eb8d8119ce3ac73329d5f7815f" target="_blank">6f1ab71</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    lectures/.DS_Store

Untracked files:
    Untracked:  GWAS.ma
    Untracked:  analysis/.Session02_practical_Key.Rmd.swp
    Untracked:  analysis/SISGM15_prac5Solution.Rmd
    Untracked:  analysis/SISGM15_prac6Solution.Rmd
    Untracked:  analysis/SISGM15_prac9Solution.Rmd
    Untracked:  analysis/Session07_practical_Key.Rmd
    Untracked:  analysis/Session08_practical_Key.Rmd
    Untracked:  causals.snplist
    Untracked:  data/sim_rels_geno.bed
    Untracked:  ldRef.bed
    Untracked:  ldRef.bim
    Untracked:  ldRef.fam
    Untracked:  ldRef.log
    Untracked:  ldRef.map
    Untracked:  ldRef.ped
    Untracked:  test1.cma.cojo
    Untracked:  test1.jma.cojo
    Untracked:  test1.ldr.cojo
    Untracked:  test1.log
    Untracked:  test2.cma.cojo
    Untracked:  test2.jma.cojo
    Untracked:  test2.ldr.cojo
    Untracked:  test2.log
    Untracked:  test3.cma.cojo
    Untracked:  test3.jma.cojo
    Untracked:  test3.ldr.cojo
    Untracked:  test3.log
    Untracked:  test4.cma.cojo
    Untracked:  test4.jma.cojo
    Untracked:  test4.ldr.cojo
    Untracked:  test4.log

Unstaged changes:
    Modified:   analysis/Session01_practical.Rmd
    Modified:   analysis/Session02_practical.Rmd
    Modified:   analysis/Session07_practical.Rmd
    Modified:   analysis/Session08_practical.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Session02_practical_Key.Rmd</code>) and HTML (<code>docs/Session02_practical_Key.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/6f1ab71addc277eb8d8119ce3ac73329d5f7815f/analysis/Session02_practical_Key.Rmd" target="_blank">6f1ab71</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
update paths
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2022_Association_Mapping/d8b2d3bc9d3537cae61ec8ec0de7d50ff7bb85c5/docs/Session02_practical_Key.html" target="_blank">d8b2d3b</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/joellembatchou/SISG2022_Association_Mapping/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/Session02_practical_Key.html" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/8b17e8c31ab2f626c3eab118ff7f6f833b4346eb/analysis/Session02_practical_Key.Rmd" target="_blank">8b17e8c</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
<td>
add key
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Before you begin:</p>
<ul>
<li>Make sure that R is installed on your computer</li>
<li>For this lab, we will use the following R libraries:</li>
</ul>
<pre class="r"><code>require(data.table)
require(dplyr)
require(tidyr)
require(bigsnpr)
require(ggplot2)</code></pre>
<div id="population-structure-inference" class="section level2">
<h2>Population Structure Inference</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>We will be working with a subset of the genotype data from the Human Genome Diversity Panel (HGDP) and HapMap.</p>
<p>The file “<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/master/data">YRI_CEU_ASW_MEX_NAM.bed</a>” is a binary file in PLINK BED format with accompanying BIM and FAM files. It contains genotype data at autosomal SNPs for:</p>
<ul>
<li>Native American samples from HGDP (NAM)</li>
<li>Four population samples from HapMap:
<ul>
<li>Yoruba in Ibadan, Nigeria (YRI)</li>
<li>Utah residents with ancestry from Northern and Western Europe (CEU)</li>
<li>Mexican Americans in Los Angeles, California (MXL)</li>
<li>African Americans from the south-western United States (ASW)</li>
</ul></li>
</ul>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>Here are some things to look at:</p>
<ol style="list-style-type: decimal">
<li>Examine the dataset:</li>
</ol>
<ul>
<li>How many samples are present?</li>
</ul>
<pre class="r"><code>famfile &lt;- fread(&quot;/data/SISG2022M15/data/YRI_CEU_ASW_MEX_NAM.fam&quot;, header = FALSE)
famfile %&gt;% head</code></pre>
<pre><code>     V1        V2 V3 V4 V5 V6
1: 1432 HGDP00702  0  0  2 -9
2: 1433 HGDP00703  0  0  1 -9
3: 1434 HGDP00704  0  0  2 -9
4: 1436 HGDP00706  0  0  2 -9
5: 1438 HGDP00708  0  0  2 -9
6: 1440 HGDP00710  0  0  1 -9</code></pre>
<pre class="r"><code>famfile %&gt;% nrow</code></pre>
<pre><code>[1] 604</code></pre>
<ul>
<li>How many SNPs?</li>
</ul>
<pre class="r"><code>bimfile &lt;- fread(&quot;/data/SISG2022M15/data/YRI_CEU_ASW_MEX_NAM.bim&quot;, header = FALSE)
bimfile %&gt;% head</code></pre>
<pre><code>   V1        V2 V3      V4 V5 V6
1:  1 rs9442372  0 1008567  1  2
2:  1 rs2887286  0 1145994  1  2
3:  1 rs3813199  0 1148140  1  2
4:  1 rs6685064  0 1201155  1  2
5:  1 rs9439462  0 1452629  1  2
6:  1 rs3820011  0 1878053  1  2</code></pre>
<pre class="r"><code>bimfile %&gt;% nrow</code></pre>
<pre><code>[1] 150872</code></pre>
<ul>
<li>What is the number of samples in each population?</li>
</ul>
<pre class="r"><code>pop_info &lt;- fread(&quot;/data/SISG2022M15/data/Population_Sample_Info.txt&quot;, header = TRUE)
head(pop_info)</code></pre>
<pre><code>    FID       IID Population
1: 1432 HGDP00702        NAM
2: 1433 HGDP00703        NAM
3: 1434 HGDP00704        NAM
4: 1436 HGDP00706        NAM
5: 1438 HGDP00708        NAM
6: 1440 HGDP00710        NAM</code></pre>
<pre class="r"><code># join with fam file
fam_pop_info &lt;- left_join(famfile, pop_info, by = c(&quot;V1&quot; = &quot;FID&quot;, &quot;V2&quot; = &quot;IID&quot;))
fam_pop_info %&gt;% select(Population) %&gt;% table</code></pre>
<pre><code>Population
ASW CEU MXL NAM YRI 
 87 165  86  63 203 </code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Get the first 10 principal components (PCs) in PLINK using all SNPs.</li>
</ol>
<pre class="r"><code>system(&quot;/data/SISG2022M15/exe/plink2 --bfile /data/SISG2022M15/data/YRI_CEU_ASW_MEX_NAM --pca 10 --out tmp/pca_out&quot;)</code></pre>
<ul>
<li>Make a scatterplot of the first two PCs with each point colored by population membership.</li>
</ul>
<pre class="r"><code>pcs &lt;- left_join(fam_pop_info, fread(&quot;tmp/pca_out.eigenvec&quot;), by = c(&quot;V1&quot; = &quot;#FID&quot;, &quot;V2&quot; = &quot;IID&quot;))
pcs %&gt;%
  ggplot(aes(x=PC1, y=PC2, color = Population)) +
  geom_point()</code></pre>
<p><img src="figure/Session02_practical_Key.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/figure/Session02_practical_Key.Rmd/unnamed-chunk-6-1.png" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Interpret the first two PCs, what ancestries are they reflecting?</li>
<li>Make a scree plot of the eigenvalues for the first 10 PCs. Approximate the proportion of variance explained by the first two PCs.</li>
</ul>
<pre class="r"><code>evals.pca &lt;- fread(&quot;tmp/pca_out.eigenval&quot;, header = FALSE)
evals.pca %&gt;%
  ggplot(aes(x = 1:10, y = V1)) +
  geom_point() +
  geom_line() +
  scale_x_continuous(breaks = 1:10) +
  labs(x = &quot;PC&quot;, y = &quot;Eigenvalue&quot;)</code></pre>
<p><img src="figure/Session02_practical_Key.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/figure/Session02_practical_Key.Rmd/unnamed-chunk-7-1.png" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sum(evals.pca$V1[1:2]) / sum(evals.pca$V1)</code></pre>
<pre><code>[1] 0.8308752</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Now redo Question 2 above using the <a href="https://privefl.github.io/bigsnpr/reference/index.html"><code>bigsnpr</code> R package</a> specifying a <span class="math inline">\(r^2\)</span> threshold of 0.2 (i.e. LD pruning) as well as a minimum minor allele count (MAC) of 20.</li>
</ol>
<ul>
<li>Run PCA and make a scatter plot of the first two principal components (PCs) with each point colored according to population membership.</li>
</ul>
<pre class="r"><code>obj.bed &lt;- bed(bedfile = &quot;/data/SISG2022M15/data/YRI_CEU_ASW_MEX_NAM.bed&quot;)
pca.bigsnpr &lt;- bed_autoSVD(
  obj.bed, 
  thr.r2 = 0.2, 
  k = 10, 
  min.mac = 20
)</code></pre>
<pre><code>
Phase of clumping (on MAC) at r^2 &gt; 0.2.. keep 87127 variants.
Discarding 48 variants with MAC &lt; 20.

Iteration 1:
Computing SVD..</code></pre>
<pre><code>The default of &#39;doScale&#39; is FALSE now for stability;
  set options(mc_doScale_quiet=TRUE) to suppress this (once per session) message</code></pre>
<pre><code>0 outlier variant detected..

Converged!</code></pre>
<pre class="r"><code>plot(pca.bigsnpr, type = &quot;scores&quot;, scores = 1:2) +
  aes(color = fam_pop_info$Population) +
  labs(color = &quot;Population&quot;)</code></pre>
<p><img src="figure/Session02_practical_Key.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/figure/Session02_practical_Key.Rmd/unnamed-chunk-8-1.png" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Does the plot change from the one in Question 2?</li>
<li>Check the SNP loadings for the first 10 PCs.</li>
</ul>
<pre class="r"><code>plot(pca.bigsnpr, type = &quot;loadings&quot;, loadings = 1:10, coeff = 0.4)</code></pre>
<p><img src="figure/Session02_practical_Key.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/figure/Session02_practical_Key.Rmd/unnamed-chunk-9-1.png" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Predict proportional Native American and European Ancestry for the HapMap MXL from the PCA output in Question 3 <em>using one of the principal components</em>. (Which PC is most appropriate for this analysis?) Assume that the HapMap MXL have negligible African Ancestry.</li>
</ol>
<pre class="r"><code>pca.bigsnpr %&gt;% str</code></pre>
<pre><code>List of 7
 $ d     : num [1:10] 2262 1434 553 495 474 ...
 $ u     : num [1:604, 1:10] 0.0488 0.043 0.0489 0.0489 0.0492 ...
 $ v     : num [1:87079, 1:10] -0.00229 -0.00165 -0.00315 -0.00888 0.00177 ...
 $ niter : num 7
 $ nops  : num 128
 $ center: num [1:87079] 0.818 0.914 0.233 0.827 0.432 ...
 $ scale : num [1:87079] 0.695 0.704 0.454 0.696 0.582 ...
 - attr(*, &quot;class&quot;)= chr &quot;big_SVD&quot;
 - attr(*, &quot;subset&quot;)= int [1:87079] 1 2 3 5 6 8 10 11 12 13 ...
 - attr(*, &quot;lrldr&quot;)=&#39;data.frame&#39;:   0 obs. of  3 variables:
  ..$ Chr  : int(0) 
  ..$ Start: int(0) 
  ..$ Stop : int(0) </code></pre>
<pre class="r"><code>ceu.mean &lt;- pca.bigsnpr$u[fam_pop_info$Population == &quot;CEU&quot;,2] %&gt;% mean
nam.mean &lt;- pca.bigsnpr$u[fam_pop_info$Population == &quot;NAM&quot;,2] %&gt;% mean
c(ceu.mean, nam.mean)</code></pre>
<pre><code>[1] -0.04885356  0.09084845</code></pre>
<pre class="r"><code>mxl.prop.nam &lt;- (pca.bigsnpr$u[fam_pop_info$Population == &quot;MXL&quot;,2] - ceu.mean) / abs(nam.mean - ceu.mean)
mxl.prop.nam %&gt;% summary</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.05262 0.37780 0.48238 0.47152 0.58381 0.81709 </code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Make a barplot of the proportional ancestry estimates from question 4.</li>
</ol>
<pre class="r"><code>data.frame(
  ind = 1:length(mxl.prop.nam), 
  NAM = sort(mxl.prop.nam, decreasing = TRUE)
  ) %&gt;%
  mutate(CEU = 1 - NAM) %&gt;%
  gather(Pop, Prop, NAM, CEU) %&gt;%
  ggplot(aes(x = ind, y = Prop, fill = factor(Pop, levels = c(&quot;NAM&quot;, &quot;CEU&quot;)))) +
  geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
  labs(x=&quot;Sample&quot;, y = &quot;Ancestry Proportion&quot;, fill = &quot;Population&quot;)</code></pre>
<p><img src="figure/Session02_practical_Key.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/figure/Session02_practical_Key.Rmd/unnamed-chunk-11-1.png" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ol start="6" style="list-style-type: decimal">
<li>Check if there are samples related 2nd degree or closer. If so, run PCA as in Question 3 removing these samples then project the remaining samples onto the PC space.</li>
</ol>
<pre class="r"><code># check for 2nd degree relateds or closer
rel.df &lt;- snp_plinkKINGQC(
  plink2.path = &quot;/data/SISG2022M15/exe/plink2&quot;, 
  bedfile.in = &quot;/data/SISG2022M15/data/YRI_CEU_ASW_MEX_NAM.bed&quot;, 
  thr.king = 2^-3.5,
  make.bed = FALSE
)
rel.df %&gt;% str</code></pre>
<pre><code>&#39;data.frame&#39;:   362 obs. of  8 variables:
 $ FID1   : chr  &quot;1563&quot; &quot;1567&quot; &quot;1567&quot; &quot;1570&quot; ...
 $ IID1   : chr  &quot;HGDP00845&quot; &quot;HGDP00849&quot; &quot;HGDP00849&quot; &quot;HGDP00852&quot; ...
 $ FID2   : chr  &quot;1556&quot; &quot;1556&quot; &quot;1561&quot; &quot;1551&quot; ...
 $ IID2   : chr  &quot;HGDP00838&quot; &quot;HGDP00838&quot; &quot;HGDP00843&quot; &quot;HGDP00832&quot; ...
 $ NSNP   : int  150801 150802 150832 150816 150822 150814 150819 150815 150674 150796 ...
 $ HETHET : num  0.0976 0.1069 0.1059 0.099 0.1022 ...
 $ IBS0   : num  0.0221 0.0219 0.0223 0.0219 0.0227 ...
 $ KINSHIP: num  0.104 0.126 0.13 0.118 0.118 ...</code></pre>
<pre class="r"><code># Gets indices of samples not related (match by fid/iid)
rel.ids &lt;- c(paste(rel.df$FID1, rel.df$IID1), paste(rel.df$FID2, rel.df$IID2))
indices.unrel &lt;- which(!(paste(famfile$V1,famfile$V2) %in% rel.ids))
indices.unrel %&gt;% str</code></pre>
<pre><code> int [1:116] 1 2 3 4 5 6 8 12 15 16 ...</code></pre>
<pre class="r"><code># Run PCA excluding relateds
pca.bigsnpr.norels &lt;- bed_autoSVD(
  obj.bed, 
  ind.row = indices.unrel,
  thr.r2 = 0.2, 
  k = 10, 
  min.mac = 20
)</code></pre>
<pre><code>
Phase of clumping (on MAC) at r^2 &gt; 0.2.. keep 77173 variants.
Discarding 12226 variants with MAC &lt; 20.

Iteration 1:
Computing SVD..
0 outlier variant detected..

Converged!</code></pre>
<pre class="r"><code># Project related samples
PCs &lt;- matrix(NA, nrow(obj.bed), ncol(pca.bigsnpr.norels$u))
PCs[indices.unrel, ] &lt;- predict(pca.bigsnpr.norels) # pc from model (unrels)
proj.rels &lt;- bed_projectSelfPCA(
  pca.bigsnpr.norels, 
  obj.bed,
  ind.row = (1:nrow(famfile))[-indices.unrel]
  )
PCs[-indices.unrel, ] &lt;- proj.rels$OADP_proj # pc from projection (rels)
data.frame(PC1 = PCs[,1], PC2 = PCs[,2], pop = fam_pop_info$Population, Type = c(&quot;Projected&quot;, &quot;Model&quot;)[1 + (1:nrow(famfile)) %in% indices.unrel]) %&gt;%
  ggplot(aes(x = PC1, y = PC2, color = pop, shape = Type)) +
  geom_point() +
  labs(color = &quot;Population&quot;)</code></pre>
<p><img src="figure/Session02_practical_Key.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/joellembatchou/SISG2022_Association_Mapping/blob/99cdf2f4424a8f7e49da9b90e93a35726992f5c4/docs/figure/Session02_practical_Key.Rmd/unnamed-chunk-12-1.png" target="_blank">99cdf2f</a>
</td>
<td>
Joelle Mbatchou
</td>
<td>
2022-07-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.6     bigsnpr_1.9.11    bigstatsr_1.5.6   tidyr_1.2.0      
[5] dplyr_1.0.9       data.table_1.14.2 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9         lattice_0.20-45    getPass_0.2-2      ps_1.7.1          
 [5] assertthat_0.2.1   rprojroot_2.0.3    digest_0.6.29      foreach_1.5.2     
 [9] utf8_1.2.2         RSpectra_0.16-1    parallelly_1.32.0  bigsparser_0.6.1  
[13] R6_2.5.1           evaluate_0.15      bigutilsr_0.3.4    highr_0.9         
[17] httr_1.4.3         pillar_1.7.0       flock_0.7          rlang_1.0.3       
[21] rstudioapi_0.13    hexbin_1.28.2      whisker_0.4        callr_3.7.0       
[25] jquerylib_0.1.4    Matrix_1.4-1       rmarkdown_2.14     labeling_0.4.2    
[29] bigparallelr_0.3.2 stringr_1.4.0      munsell_0.5.0      compiler_4.2.1    
[33] httpuv_1.6.5       xfun_0.31          pkgconfig_2.0.3    htmltools_0.5.2   
[37] tidyselect_1.1.2   tibble_3.1.7       codetools_0.2-18   viridisLite_0.4.0 
[41] fansi_1.0.3        withr_2.5.0        crayon_1.5.1       later_1.3.0       
[45] grid_4.2.1         jsonlite_1.8.0     gtable_0.3.0       lifecycle_1.0.1   
[49] DBI_1.1.3          git2r_0.30.1       magrittr_2.0.3     scales_1.2.0      
[53] cli_3.3.0          stringi_1.7.6      farver_2.1.1       fs_1.5.2          
[57] promises_1.2.0.1   doParallel_1.0.17  robustbase_0.95-0  bslib_0.3.1       
[61] ellipsis_0.3.2     generics_0.1.3     vctrs_0.4.1        cowplot_1.1.1     
[65] iterators_1.0.14   tools_4.2.1        glue_1.6.2         DEoptimR_1.0-11   
[69] purrr_0.3.4        processx_3.7.0     parallel_4.2.1     fastmap_1.1.0     
[73] yaml_2.3.5         colorspace_2.0-3   bigassertr_0.1.5   bigreadr_0.2.4    
[77] knitr_1.39         sass_0.4.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
