---
title: "Module 15 - Association mapping: GWAS & sequencing data"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

```{r echo = FALSE, message=FALSE, warning=FALSE}
require(dplyr)
```

Instructors: Joelle Mbatchou & Loic Yengo

You will find on this website link to lecture slides, exercises including some solutions, and link to the class video recordings.

The official SISG web page can be found [here](https://si.biostat.washington.edu/courses/SM2215) (requires login) â€“ course evaluations and certificates of completion are available via this page.


## Session Format
The module has 10 sessions, each of 90 minutes. The standard format for a session is approximately:

* 45 minutes of lecture material that will be recorded via Zoom live and posted at the end of the day
* 30 minutes of exercises for you to try, with small-group "breakout" Zoom sessions available, attended by other class participants, and Teaching Assistants
* 15 minute discussion of exercises, where the instructors will present possible solutions and answer questions

Please join the module's [Slack channel](https://uwbiostatisticssisg.slack.com), where you can ask questions and see real-time updates from the instructors and TAs.
Each session builds on previous sessions so you will get maximum benefit by attending every session live and in sequence.

The course schedule is below.


## Pre-requisites

**Prior to the module**, please install working and latest versions of:

* R (Version 4.1+): 
  <a href="https://www.r-project.org/" target="_blank">https://www.r-project.org/</a>
* RStudio (Version 1.4+) : 
  <a href="https://rstudio.com/products/rstudio/download/" target="_blank">https://rstudio.com/products/rstudio/download/</a>
* PLINK 1.9 : 
  <a href="https://www.cog-genomics.org/plink2/" target="_blank">https://www.cog-genomics.org/plink2/</a>
* PLINK 2.0 : 
  <a href="https://www.cog-genomics.org/plink2.0/" target="_blank">https://www.cog-genomics.org/plink2.0/</a>
* REGENIE : 
  <a href="https://github.com/rgcgithub/regenie/releases/" target="_blank">https://github.com/rgcgithub/regenie/releases/</a>
* Zoom : 
  <a href="https://zoom.us/" target="_blank">https://zoom.us/</a>

All are free software. 
REGENIE requires MAC/Linux system; if you have a Windows laptop, you can use docker to setup a Linux image (see guide here).

Please be sure to install some necessary R packages with:
```{r install-pkgs, eval = FALSE}
install.packages("qqman")
install.packages("bigsnpr")
install.packages("hexbin")
install.packages("tidyverse")
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("GWASTools")
```

## Schedule

All times listed below for the schedule are Pacific Standard Time (PST, Seattle time).

```{r mon-sched, echo = FALSE, results = "asis"}
data.frame(
  Time = c("8:00am-9:30am", "10am-11:30am", "12pm-1:30pm", "1:45pm-2:30pm"),
  Session = 1:4,
  Topic = c("Introduction, Case Control Association Testing & Association Testing with Quantitative Traits",
    "Introduction to the PLINK Software for GWAS & Population Structure Inference",
    "GWAS in Samples with Structure & Introduction to the REGENIE Software",
    "Power, Sample Size & Design Considerations"
    ),
  Lecture = c(
    "[Intro](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session00.pdf), [Lecture](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session01.pdf), Video",
    "[Lecture](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session03.pdf), Video",
    "[Lecture](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session06.pdf), Video",
    "[Lecture](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session10.pdf), Video" 
      ),
  "Exercises" = c(
    "[Exercises](Session01_practical.html), Video",
    "[Exercises](Session02_practical.html), Video",
    "[Exercises](Session03_practical.html), Video",
    "[Exercises](https://faculty.washington.edu/tathornt/SISG2021/exercises/assoc_ex4.pdf), Video"
  ),
  "Key" = c(
     "[.html], [.Rmd]",
    "[.html], [.Rmd]",
    "[.html], [.Rmd]",
    "[.html], [.Rmd]"
  )
 , check.names = FALSE ) %>%
knitr::kable("html", escape = FALSE, 
  caption = "Monday, July 25th", 
  align = c('l', 'c', 'l', 'l', 'l')
) %>%
kableExtra::kable_styling(bootstrap_options = c("hover", "condensed","striped"))
```

```{r tue-sched, echo = FALSE, results = "asis"}
data.frame(
  Time = c("8:00am-9:30am", "10am-11:30am", "12pm-1:30pm", "1:45pm-2:30pm"),
  Session = 4 + 1:4,
  Topic = c("Downstream analyses: conditional analyses, colocalization & fine-mapping",
    "Downstream analyses: gene-based and pathways analysis & polygenic risk scores",
    "Rare Variant Analysis: Collapsing Tests, Kernel (Variance Component) Tests and Omnibus Tests",
    "Emerging issues showcasing ongoing research"
    ),
  Lecture = c(
    "[Intro](TBD), [Lecture](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session01.pdf), Video",
    "[Lecture](TBD), Video",
    "[Lecture](TBD), Video",
    "[Lecture](TBD), Video" 
      ),
  "Exercises" = c(
    "[Exercises](TBD), Video",
    "[Exercises](TBD), Video",
    "[Exercises](Session07_practical.html), Video",
    "[Exercises](TBD), Video"
  ),
  "Key" = c(
    "[.html ], [.Rmd]",
    "[.html ], [.Rmd]",
    "[.html ], [.Rmd]",
    "[.html ], [.Rmd]"
  )
  , check.names = FALSE) %>%
knitr::kable("html", escape = FALSE, 
  caption = "Tuesday, July 26th", 
  align = c('l', 'c', 'l', 'l', 'l')
) %>%
kableExtra::kable_styling(bootstrap_options = c("hover", "condensed","striped"))
```

```{r wed-sched, echo = FALSE, results = "asis"}
data.frame(
  Time = c("8:00am-9:30am", "10am-11am"),
  Session = 8 + 1:2,
  Topic = c("Beyond traditional GWAS: Variance QTL, Interaction Testing, GWAX, Time-to-event, Multi-trait analysis",
    "Emerging Issues showcasing ongoing research"
    ),
  Lecture = c(
    "[Intro](TBD), [Lecture](https://faculty.washington.edu/tathornt/SISG2021/lectures/SISG2021session01.pdf), [Video](link4)",
    "[Lecture](TBD), Video" 
      ),
  "Exercises" = c(
    "[Exercises](TBD), Video",
    "[Exercises](TBD), Video"
  ),
  "Key" = c(
    "[.html ], [.Rmd]",
    "[.html ], [.Rmd]"
  )
  , check.names = FALSE) %>%
knitr::kable("html", escape = FALSE, 
  caption = "Wednesday, July 27th", 
  align = c('l', 'c', 'l', 'l', 'l')
) %>%
kableExtra::kable_styling(bootstrap_options = c("hover", "condensed","striped"))
```

## Datasets

All individual data files below can be downloaded from a zipped folder in Dropbox.

`r #Alternatively, you can download the files from the Github repository (the largest file "Transferrin.bed" can be downloaded from the Dropbox folder above).`

## Additional resources

* [2009 AJHG article](https://faculty.washington.edu/tathornt/sisg/AJHG_Transferrin_2009.pdf) by Benyamin et al for a GWAS of Transferrin
* [2016 TPB Paper](https://faculty.washington.edu/tathornt/sisg/Zheng_Weir_Eigenanalysis_TPB_2016.pdf) on EIGENANALYSIS (PCA) for population Structure Inference and Admixture Estimation
* [HapMap MXL Individual Ancestry Figure](https://faculty.washington.edu/tathornt/sisg/HapMapMXL_Ancestry_Supervised.pdf) (Supervised Learning)
* If you are new to R, you may find the following material helpful:
  * [Introduction to R materials](https://faculty.washington.edu/kenrice/rintro/) from SISG Module 3
  * [Graphics with ggplot2](https://ggplot2.tidyverse.org/)
  * [Data manipulation with dplyr](http://dplyr.tidyverse.org/)
