---
title: "Module 15 - Association mapping: GWAS & sequencing data"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r echo = FALSE, message=FALSE, warning=FALSE}
require(dplyr)
```

Instructors: Loic Yengo & Joelle Mbatchou

You will find on this website link to lecture slides, exercises including some solutions, and link to the class video recordings.


## Session Format
The module has 10 sessions, each of 90 minutes. The standard format for a session is approximately:

* 45 minutes of lecture material that will be recorded via Zoom live and posted at the end of the day
* 30 minutes of exercises for you to try, with small-group "breakout" Zoom sessions available, attended by other class participants, and Teaching Assistants
* 15 minute discussion of exercises, where the instructors will present possible solutions and answer questions

Please join the module's [Slack channel](TBD), where you can ask questions and see real-time updates from the instructors and TAs.
Each session builds on previous sessions so you will get maximum benefit by attending every session live and in sequence.

The tentative course schedule is below.


## Pre-requisites

**Prior to the module**, please install working and latest versions of:

* R (Version 4.1+): 
  <a href="https://www.r-project.org/" target="_blank">https://www.r-project.org/</a>
* RStudio (Version 1.4+) : 
  <a href="https://rstudio.com/products/rstudio/download/" target="_blank">https://rstudio.com/products/rstudio/download/</a>
* PLINK 1.9 : 
  <a href="https://www.cog-genomics.org/plink2/" target="_blank">https://www.cog-genomics.org/plink2/</a>
* PLINK 2.0 : 
  <a href="https://www.cog-genomics.org/plink2.0/" target="_blank">https://www.cog-genomics.org/plink2.0/</a>
* REGENIE : 
  <a href="https://github.com/rgcgithub/regenie/releases/" target="_blank">https://github.com/rgcgithub/regenie/releases/</a>
* Zoom : 
  <a href="https://zoom.us/" target="_blank">https://zoom.us/</a>

All are free software. 
REGENIE requires MAC/Linux system; if you have a Windows laptop, you can use docker to setup a Linux image (see guide here).

Please be sure to install some necessary R packages with 
```{r install-pkgs, eval = FALSE}
install.packages("qqman")
install.packages("GWASTools")
install.packages("bigsnpr")
install.packages("ggplot2")
```

## Schedule

This is a tentative schedule. All times listed below for the schedule are Pacific Standard Time (Seattle time, PST).

```{r echo = FALSE, results = "asis"}
data.frame(
  Time = c("8:00am-9:20am", "9:40am-11:00am"),
  Session = 1:2,
  Topic = c("Introduction, Case Control Association Testing",
    "Association Testing with Quantitative Traits"),
  Lecture = c(
    "[Intro](link), [Lecture](link), [Video](link4)",
    "[Lecture](link3), [Video](link4)" 
      ),
  "Exercises" = c(
    "[Exercises](.pdf), Video",
    "[Exercises](.pdf), Video"
  ),
  "Solution key" = c(
    "Key: [.html ], [.Rmd]",
    "Key: [.html ], [.Rmd]"
  )
  ) %>%
knitr::kable("html", escape = FALSE, 
  caption = "Monday, July 25th", 
  align = c('l', 'c', 'l', 'l', 'l')
) %>%
kableExtra::kable_styling(bootstrap_options = c("hover", "condensed","striped"))
```


## Datasets

All individual data files below can be downloaded from two zipped folder from dropbox.

Alternatively, you can download each of the data files in the Github repository.

## Additional resources

* A tutorial for the SNPRelate package is available here: SNPRelate Tutorial
* 2009 AJHG article by Benyamin et al for a GWAS of Transferrin
* 2016 TPB Paper on EIGENANALYSIS (PCA) for population Structure Inference and Admixture Estimation
* HapMap MXL Individual Ancestry Figure (Supervised Learning)
