---
title: "Session 03 - Exercises Key"
author: ""
date: ""
output: html_document
editor_options:
  chunk_output_type: console
---
```{r echo = FALSE}
knitr::opts_knit$set(root.dir = "~/")
```
Before you begin:

* Make sure that R is installed on your computer
* For this lab, we will use the following R libraries:
```{r load-libs, message=FALSE, warning=FALSE}
require(data.table)
require(dplyr)
require(tidyr)
require(GWASTools)
require(ggplot2)
```

## GWAS in Samples with Structure

### Introduction
We will be analyzing a simulated data set which contains sample structure to better understand the impact it can have in GWAS analyses if not accounted for. 
We will perform GWAS on a quantitative phenotype which was simulated to have high heritability and be highly polygenic.

The file ["sim_rels_pheno.txt"](https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/master/data)‚Äù contains the phenotype measurements 
for a set of individuals and the file "sim_rels_geno.bed" is a binary file in PLINK BED format 
with accompanying BIM and FAM files which contains the genotype data **at null variants** (i.e. simulated as not associated with the phenotype).  
*How should we expect the QQ/Manhatthan plots to look like under this scenario?*

### Exercises
Here are some things to try:

  1. Examine the dataset:
  * How many samples are present? 
```{r}
famfile <- fread("/data/SISG2022M15/data/sim_rels_geno.fam", header = FALSE)
famfile %>% str
```
  * How many SNPs? In how many chromosomes?
```{r}
bimfile <- fread("/data/SISG2022M15/data/sim_rels_geno.bim", header = FALSE)
bimfile %>% str
bimfile %>% select(V1) %>% table
```

  2. Examine the phenotype data:
  * How many individuals in the study have measurements?
```{r}
yfile <- fread("/data/SISG2022M15/data/sim_rels_pheno.txt", header = TRUE)
yfile %>% str
yfile %>% pull(Pheno) %>% is.na %>% table
```
  * Make a visual of the distribution of the phenotype?
```{r}
yfile %>%
  ggplot(aes(x = Pheno)) +
  geom_histogram(colour="black", fill="white")
```

3. Using PLINK, perform a GWAS using the phenotype file `sim_rels_pheno.txt` and the `sim_rels_geno.{bed,bim,fam}` genotype files.
Only perform association test on SNPs that pass the following quality control threshold filters:

* minor allele frequency (MAF) > 0.01
* at least a 99% genotyping call rate (less than 1% missing)
* HWE p-values greater than 0.001

```{r}
system("/data/SISG2022M15/exe/plink2 --bfile /data/SISG2022M15/data/sim_rels_geno --pheno /data/SISG2022M15/data/sim_rels_pheno.txt --pheno-name Pheno --maf 0.01 --geno 0.01 --hwe 0.001 --autosome --glm allow-no-covars --out tmp/gwas_plink")
```

4. Make a Manhattan plot of the association results using the `manhattanPlot()` R function.
```{r}
plink.gwas <- fread("tmp/gwas_plink.Pheno.glm.linear", header = TRUE)
plink.gwas %>% str
manhattanPlot(
  p = plink.gwas$P,
  chromosome = plink.gwas$`#CHROM`, 
  thinThreshold = 1e-4,
  main= "Manhattan plot of GWAS with PLINK"
)
```

5. Make a Q-Q plot of the association results using the `qqPlot()` R function.
```{r}
qqPlot(
  pval = plink.gwas$P,
  thinThreshold = 1e-4,
  main= "Q-Q plot of GWAS with PLINK"
 )
```

6. Compute the genomic control inflation factor $\lambda_{GC}$ based on the p-values.
Is there evidence of possible inflation due to confounding?
```{r}
chisq.stats <- qchisq(plink.gwas$P, df = 1, lower.tail = FALSE)
median(chisq.stats) / qchisq(0.5,1)
```

7. Now use REGENIE to perform a GWAS of the phenotype using a whole genome regression model.

  * We want to use high quality variants in the Step 1 null model fitting.
  Using PLINK, apply QC filters to remove variants with MAF below 5%, missingness above 1%, HWE p-value below 0.001, minor allele count (MAC) below 20. 
```{r}
system("/data/SISG2022M15/exe/plink2 --bfile /data/SISG2022M15/data/sim_rels_geno --maf 0.05 --geno 0.01 --hwe 0.001 --mac 20 --write-snplist --out tmp/qc_pass")
```

  * Run REGENIE Step 1 to fit the null model and obtain polygenic predictions using a leave-one-chromosome-out (LOCO) scheme
```{r, echo = TRUE, results='hide'}
system("/data/SISG2022M15/exe/regenie --bed /data/SISG2022M15/data/sim_rels_geno --phenoFile /data/SISG2022M15/data/sim_rels_pheno.txt --step 1 --loocv --bsize 1000 --qt --extract tmp/qc_pass.snplist --out tmp/regenie_step1")
```

The prediction list file output from Step 1 contains the path to the LOCO polygenic predictions:
```{r}
fread("tmp/regenie_step1_pred.list", header = FALSE)
```

  * Run REGENIE Step 2 to perform association testing **at the same set of SNPs tested in PLINK**.
```{r}
plink.gwas %>%
  select(ID) %>%
  fwrite("tmp/plink_gwas.snplist", col.names = FALSE, quote = FALSE)
```

```{r, echo = TRUE, results='hide'}
system("/data/SISG2022M15/exe/regenie --bed /data/SISG2022M15/data/sim_rels_geno --phenoFile /data/SISG2022M15/data/sim_rels_pheno.txt --step 2 --bsize 400 --qt --pred tmp/regenie_step1_pred.list --extract tmp/plink_gwas.snplist --out tmp/regenie_step2")
```  

  * Generate Manhatthan and Q-Q plots based on the association results and compute $\lambda_{GC}$. Compare with output from Questions 4-6.
```{r}
regenie.gwas <- fread("tmp/regenie_step2_Pheno.regenie", header = TRUE)
regenie.gwas %>% str
manhattanPlot(
  p = 10^-regenie.gwas$LOG10P,
  chromosome = regenie.gwas$CHROM, 
  thinThreshold = 1e-4,
  main= "Manhattan plot of GWAS with REGENIE"
)
qqPlot(
  pval = 10^-regenie.gwas$LOG10P,
  thinThreshold = 1e-4,
  main= "Q-Q plot of GWAS with REGENIE"
 )
chisq.stats <- qchisq(10^-regenie.gwas$LOG10P, df = 1, lower.tail = FALSE)
median(chisq.stats) / qchisq(0.5,1)
```
