---
title: "Session 03 - Exercises"
author: ""
date: ""
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
Before you begin:

* Make sure that R is installed on your computer
* For this lab, we will use the following R libraries:
```{r load-libs, eval=FALSE}
require(data.table)
require(dplyr)
require(tidyr)
require(GWASTools)
require(ggplot2)
```

The R template to do the exercises is [here](https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/master/code).

*Note: if on the online server, set your working directory to your home directory using in R*
```{r, eval = FALSE} 
setwd("home/<username>/")
```

The data files are in the folder `/data/SISG2022M15/data/`.


## GWAS in Samples with Structure & Using REGENIE

### Introduction
We will be analyzing a simulated data set which contains sample structure to better understand the impact it can have in GWAS analyses if not accounted for. 
We will perform GWAS on a quantitative phenotype which was simulated to have high heritability and be highly polygenic.

The file ["sim_rels_pheno.txt"](https://github.com/joellembatchou/SISG2022_Association_Mapping/tree/master/data)â€ contains the phenotype measurements 
for a set of individuals and the file "sim_rels_geno.bed" is a binary file in PLINK BED format 
with accompanying BIM and FAM files which contains the genotype data **at null variants** (i.e. simulated as not associated with the phenotype).  
*How should we expect the QQ/Manhatthan plots to look like under this scenario?*

### Exercises
Here are some things to try:

1. Examine the dataset:
  * How many samples are present? 
  * How many SNPs? In how many chromosomes?

2. Examine the phenotype data:
  * How many individuals in the study have measurements?
  * Make a visual of the distribution of the phenotype.

3. Using PLINK, perform a GWAS using the phenotype file `sim_rels_pheno.txt` and the `sim_rels_geno.{bed,bim,fam}` genotype files.
Only perform association test on SNPs that pass the following quality control threshold filters:

* minor allele frequency (MAF) > 0.01
* at least a 99% genotyping call rate (less than 1% missing)
* HWE p-values greater than 0.001

The basic command would look like
```{bash, eval = FALSE}
plink2 --bfile sim_rels_geno --pheno sim_rels_pheno.txt --pheno-name <pheno_name> --maf <min_MAF> --geno <max_miss> --hwe <hwe_p_thresh> --glm allow-no-covars --out <output_prefix>
```

4. Make a Manhattan plot of the association results using the `manhattanPlot()` R function. The basic command would look like
```{r, eval = FALSE}
manhattanPlot(
  p = <pvalues>,
  chromosome = <chromosomes>, 
  thinThreshold = 1e-4,
  main= <title>
)
```

5. Make a Q-Q plot of the association results using the `qqPlot()` R function. The basic command would look like
```{r, eval = FALSE}
qqPlot(
  pval = <pvalues>,
  thinThreshold = 1e-4,
  main= <title>
 )
```

6. Compute the genomic control inflation factor $\lambda_{GC}$ based on the p-values.
(Hint: convert p-values to $\chi^2_1$ test statistics using the R function `qchisq()`). 
Is there evidence of possible inflation due to confounding?

7. Now use REGENIE to perform a GWAS of the phenotype
using a whole genome regression model.
  
  * We want to use high quality variants in the Step 1 null model fitting.
  Using PLINK, apply QC filters to remove variants with MAF below 5%, missingness above 1%, HWE p-value below 0.001, minor allele count (MAC) below 20. 
  (hint: use `--write-snplist` to store list of variants passing QC without making a new BED file)
  
  * Run REGENIE Step 1 to fit the null model and obtain polygenic predictions using a leave-one-chromosome-out (LOCO) scheme. The basic command would look like
```{bash, eval = FALSE}
regenie --bed sim_rels_geno --phenoFile sim_rels_pheno.txt --step 1 --loocv --bsize 1000 --qt --extract <plink_QC_pass_snplist> --out <output_prefix_step1>
```

  * Run REGENIE Step 2 to perform association testing **at the same set of SNPs tested in PLINK**. The basic command would look like
```{bash, eval = FALSE}
regenie --bed sim_rels_geno --phenoFile sim_rels_pheno.txt --step 2 --bsize 400 --qt  --pred <output_prefix_step1>_pred.list --extract <plink_GWAS_snplist> --out <output_prefix_step2>
```  

  * Generate Manhatthan and Q-Q plots based on the association results and compute $\lambda_{GC}$. Compare with output from Questions 4-6.
  
